NitroxPatcher/Patches/Dynamic/PingInstance_SetColor_Patch.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Reflection;
+using HarmonyLib;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+using UnityEngine;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+/// <summary>
+/// Notice the server of a Ping color change
+/// </summary>
+public class PingInstance_SetColor_Patch : NitroxPatch, IDynamicPatch
+{
+    private static readonly MethodInfo TARGET_METHOD = Reflect.Method((PingInstance t) => t.SetColor(default));
+
+    public static void Prefix(PingInstance __instance, int index)
+    {
+        // Only notice server if there's a change on client-side
+        if (__instance.colorIndex == index)
+        {
+            return;
+        }
+        
+        if (TryGetPingKey(__instance.gameObject, out string pingKey))
+        {
+            Resolve<IPacketSender>().Send(new SignalPingPreferenceChanged(pingKey, index));
+        }
+    }
+
+    // We need to differentiate two types of pings, the "normal pings" from objects that emit a signal, these objects generally contain a NitroxEntity
+    // And the other type is Signal pings that are generated by the story events, they are located in the Global Root and don't contain a NitroxEntity, to be identified, they have another object: a SignalPing which contains a description key
+    public static bool TryGetPingKey(GameObject gameObject, out string pingKey)
+    {
+        pingKey = string.Empty;
+        if (!NitroxEntity.TryGetEntityFrom(gameObject, out NitroxEntity entity))
+        {
+            if (gameObject.TryGetComponent(out SignalPing signalPing))
+            {
+                pingKey = signalPing.descriptionKey;
+            }
+            else
+            {
+                Log.Warn($"Couldn't find PingInstance identifier for {gameObject.name}");
+            }
+        }
+        else
+        {
+            pingKey = entity.Id.ToString();
+        }
+        return pingKey != string.Empty;
```

Comment:
Didn't test this ingame:
```suggestion
        if (NitroxEntity.TryGetEntityFrom(gameObject, out NitroxEntity entity))
        {
            pingKey = entity.Id.ToString();
            return true; // Id of a NitroxEntity should never be null.
        }
        
        if (gameObject.TryGetComponent(out SignalPing signalPing))
        {
            pingKey = signalPing.descriptionKey;
            return !string.IsNullOrEmpty(pingKey);
        }
        
        Log.Warn($"Couldn't find PingInstance identifier for {gameObject.name}");
        pingKey = string.Empty;
        return false;
```